### ISSUE TITLE
PROBLEM TO CREATE TABLE FOR

### ISSUE THREAD

**Issue by DexterWinklerGroening** [sqlitebrowser/sqlitebrowser#3286]
The problem is that the second table cannot be created, inclusive I tried to drop the first table and the problem continues.
![Captura de pantalla 2023-02-26 180305](https://user-images.githubusercontent.com/126422407/221440201-4ee762bc-3dcb-4918-859d-034321e96b2a.png)

**Comment by lucydodo** [comment]
Well, I just created a new database and entered the same SQL code, but was unable to reproduce the issue.
Are you sure you don't have User table? Try this again: `DROP TABLE User;`

**Comment by justinclift** [comment]
@DexterWinklerGroening Also, which version of our application are you using, and on what operating system? :smile:

**Comment by chrisjlocke** [comment]
Everyone else is joining in, so would be rude for me not to.  Once you've answered everyone else's questions, feel free to answer mine! ;)

![image](https://user-images.githubusercontent.com/7872218/221565774-f431daf6-d4c8-45d5-9510-39d30da5c6cb.png)

The error is says (1) is 'user already exists'.  This means the table you're trying to create already exists.

Now, you can check this in two ways.  The application you're using is a GUI, so it shows things to the user graphically.  However, the part of the application you're in (execute SQL) is the 'command line' part of the application.  Useful for speed and doing repetitive tasks.

So, lets use the GUI first.  It's a bit easier.

![image](https://user-images.githubusercontent.com/7872218/221566684-a6663645-a446-4679-8698-b1dff69a39a6.png)

Click on the 'Database Structure' tab (1) and you should see a list of tables.  I've created the above tables, so you might see more or less than here.  You can see one says 'User' (2)

**Deleting a table**

From this screen you can delete a table.  As you're trying to create it, you can delete it from here.

![image](https://user-images.githubusercontent.com/7872218/221567091-65a61fa2-a2c5-447b-a505-4da51c02b217.png)

Click the 'User' table (1) once to select it, then press 'Delete table' (2)

![image](https://user-images.githubusercontent.com/7872218/221567163-bd603ffd-c077-48a1-b502-985bacb39bc0.png)

You'll be asked to confirm this action.  Click 'Yes'.

**Creating the table via SQL**

![image](https://user-images.githubusercontent.com/7872218/221567587-9226c3d0-748a-43f5-9847-5f191157c67d.png)

Select the 'Execute SQL' tab (1) and enter the required SQL (2).  Press the 'Execute' button (3) or you can press [F5]

![image](https://user-images.githubusercontent.com/7872218/221567849-e619990d-c727-4567-9b80-be846e7caccd.png)

If this works, you'll see this result - 'finished without errors'.  You can go back to the 'Database structure' tab to confirm the table is there.

**Creating a table using the GUI**

While you have SQL to create the user table, you can use the GUI as well.

![image](https://user-images.githubusercontent.com/7872218/221568987-749091a6-6a11-4d66-b871-df6473924e4b.png)

Click the 'Database structure' tab (1) and click 'Create table' (2)

![image](https://user-images.githubusercontent.com/7872218/221569235-3a24afda-b0bd-4bcf-b5b9-5a3d48257d07.png)

In the 'table' field (1), enter 'Users'.  Click 'Add' (2) to add a new field.  Enter the name 'id' (3), make sure the type is 'integer' and click all the clickables (4).

![image](https://user-images.githubusercontent.com/7872218/221569684-d96fb27a-8d05-4b9f-b14c-c3cc0f0390ad.png)

Repeat the process for the remaining two fields - press 'Add' (1), enter the field name (2) and change the type to 'text'.  Click 'OK' at the bottom of the screen to confirm.

![image](https://user-images.githubusercontent.com/7872218/221570042-a57e7378-ecf4-4333-b7f4-dded9a6bfcab.png)

Once done, as per previously, you'll see the tables in your database.


Part two of this is adding records into the tables and viewing these to make sure you've entered them correctly.

**Comment by chrisjlocke** [comment]
Appreciate this maybe isn't your code, but setting 'not null, autoincrement' and 'unique' for the primary key is unnecessary.  I'll wobble on about what each setting is for, so feel free to ignore me.  Most people do.

**Not null**
This means you cannot set the field to nothing.  if you're creating a numeric primary key, then this is a special field as far as sqlite is concerned, and cannot be set to nothing anyway ... so its not needed.

**Auto increment**
Normally when you add records but then delete them, sqlite can possibly reuse the record numbers again.  

![image](https://user-images.githubusercontent.com/7872218/221572279-4b1c1e32-5490-4b71-9f1e-83d629fc2811.png)

So if I create a table and add 5 records to it, I can see all 5 records.

![image](https://user-images.githubusercontent.com/7872218/221572474-f839ebaa-4980-4c09-9590-a68488f3141c.png)

If I delete two records...

![image](https://user-images.githubusercontent.com/7872218/221572531-07162c53-b023-4f2c-865c-f03857a17b59.png)

But then add two more records, it uses record number 4 and record number 5 again.

![image](https://user-images.githubusercontent.com/7872218/221572882-85ce7fa6-5e20-428c-b013-4b3feceba388.png)

This won't work if I delete record number 3.  If I delete record 3 and add a new record, it'll create record 6.
It only reuses records that were at the end of the table and can be reused.

Setting the 'auto increment' field removes this ability - record numbers are never reused.

**unique**

The 'Unique' flag means that the field contents can't match the same field in another record in the same table .. so you can't have two records with '4' in the field, for example.  Again, as an integer marked as 'primary key' has special meaning to sqlite, the 'unique' is implied - you cannot have non-unique values in this field even if you wanted them.

So in conclusion, when setting a primary key, all thats really needed is 'integer primary key'.  The 'auto increment' is personal preference, but 98.54% of the time, leaving it off is the norm.
